lib/services/api/api_service.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:survey/gen/configs.gen.dart';
+import 'package:survey/services/http/http_service.dart';
+import 'package:object_mapper/object_mapper.dart';
+import 'package:survey/services/locator/locator_service.dart';
+
+part 'api_exception.dart';
+
+part 'api_params.dart';
+
+part 'api_service_register.dart';
+
+abstract class ApiService {
+  Future<T> call<T extends Mappable>({
+    required HttpMethod method,
+    String? baseUrl,
+    required String endPoint,
+    ApiParams? params,
+    String? token,
+  });
```

Comment:
@markgravity are you missing `requiresAuthentication` here? 🤪

Reply:
@manh-t my bad. Updated 👍